orangecontrib/text/pubmed.py

```diff
@@ -0,0 +1,480 @@
+                    .format(type(terms)),
+                'conjunction': ' OR ',
+                'field': 'All fields',
+            },
+        ]
+
+        for check in argument_checks:
+            if not isinstance(check.get('argument'), list):
+                raise ValueError(check.get('error'))
+            arguments_list = ['{}[{}]'.format(arg, check.get('field'))
+                              for arg in check.get('argument')]
+            arguments = check.get('conjunction').join(arguments_list)
+            if arguments:
+                search_terms.append('({})'.format(arguments))
+
+        # The 'retmax' parameter determines the maximum number of record ids
+        # we want returned, where 100000 is the maximum. Setting it to maximum
+        # will therefore return all available record ids.
+        try:
+            search_handle = Entrez.esearch(
+                    db='pubmed',
+                    term=' AND '.join(search_terms),
+                    mindate=pub_date_start,
+                    maxdate=pub_date_end,
+                    datetype='pdat',
+                    usehistory='y',
+                    retmax='100000'
+            )
+        except Exception as e:
+            warnings.warn(
+                'An exception occurred ({0}).'.format(e),
+                RuntimeWarning
+            )
+            if self.error_callback:
+                self.error_callback(e)
+            return
+
+        # Read into a dictionary.
+        search_results = Entrez.read(search_handle)
+        search_handle.close()
+
+        # Store data from this search.
+        self.record_id_list = search_results.get('IdList')
+        self.search_record_count = int(search_results.get('Count'))
+        self.search_record_web_env = search_results.get('WebEnv')
+        self.search_record_query_key = search_results.get('QueryKey')
+
+    def _search_for_records_advanced(self, query):
+        """Executes a advanced/custom search for PubMed records.
+
+        The query is in the same format as the one assembled on the PubMed
+        website. Results of the search, like QueryKey, WebEnv and Count are
+        stored as parameters of this class's instance.
+
+        Args:
+            query (str): The advanced query string.
+        """
+        if not query:
+            warnings.warn('Cannot run PubMed query on empty input.',
+                          RuntimeWarning)
+            return
+
+        try:
+            search_handle = Entrez.esearch(
+                    db='pubmed',
+                    term=query,
+                    usehistory='y',
+                    retmax='100000'
+            )
+        except Exception as e:
+            warnings.warn(
+                'An exception occurred ({0}).'.format(e),
+                RuntimeWarning
+            )
+            if self.error_callback:
+                self.error_callback(e)
+            return
+        search_results = Entrez.read(search_handle)
```

Comment:
The following 8 lines are duplicated and the method is in general quite similar to `_search_for_records`. Could `_search_for_records` just construct an advanced query and pass it to this method?